### DISCUSSION TITLE
firmware for Seeed Studio XIAO ESP32C3

### DISCUSSION THREAD

**Discussion by uwe5** [arendst/Tasmota#16835] [Q&A] [answered]
Which firmware has to be loaded on the "Seeed Studio XIAO ESP32C3"?
tasmota32c3cdc.bin or tasmota32c3.bin

And what is the difference between tasmota32c3cdc.bin and tasmota32c3.bin?

Many thanks in advance

**Comment by barbudor** [comment, marked as answer]
As the SeeedStudio XIAO doesn't have a separated USB to Serial chip and leverage the build-in C3's CDC, you should go for tasmota32c3cdc.bin

Tiny little jewel this board 😮

**Reply by sfromis** [reply]
"CDC" for "USB Communication Device Class" which gives you a serial-like console over USB.

I'm using it with a board of similar size, but with 5x5 neopixels on one side. With a small Berry program, great for locating a device I had forgotten where was, turning up colors/brightness/pixel number to indicate signal strength as I walked around.

**Reply by uwe5** [reply]
I completely recompiled Tasmota and loaded it directly via USB with PlatformIO. **Works great,** with the external antenna the tiny-thing has a super WLAN signal.

To do this I took the following steps.

1. Create a "99-platformio-udev.rules" file under /etc/udev/rules.d/ and fill it with the content:

```
# Espressif USB JTAG/serial debug unit
ATTRS{idVendor}=="303a", ATTRS{idProduct}=="1001", MODE="0666", ENV{ID_MM_DEVICE_IGNORE}="1", ENV{ID_MM_PORT_IGNORE}="1"

```

2. copied platformio_override_sample.ini to platformio_override.ini and fill it with:

```
[platformio]
default_envs            = env:tasmota32c3cdc-DE

[env:tasmota32c3cdc-DE]
extends                 = env:tasmota32c3cdc
upload_port             = /dev/ttyACM0
build_flags             = ${env:tasmota32c3cdc.build_flags} -DMY_LANGUAGE=de_DE
```
3. let everything compile and flash.

4. in the PlatformIO Serial Monitor you can see this:

16:18:26.804 APP: starte neu
ESP-ROM:esp32c3-api1-20210207
Build:Feb  7 2021
rst:0x3 (RTC_SW_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40382e2c
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd5810,len:0x48
load:0x403cc710,len:0x64c
load:0x403ce710,len:0x216c
SHA-256 comparison failed:
Calculated: e0d75ff7e0926df9f88222e5bdef5d235358e8deede65e96e9b562dc3b033d94
Expected: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
Attempting to boot anyway...
entry 0x403cc710

00:00:00.001 CMD: Using USB CDC
00:00:00.001 HDW: ESP32-C3 
00:00:00.009 UFS: FlashFS mounted with 304 kB free
00:00:00.020 CFG: Loaded from File, Anzahl 16
00:00:00.025 QPC: Count 1
00:00:00.064 BRY: Berry initialized, RAM used=4415 bytes
00:00:00.074 Projekt tasmota - Seeed-Studio-XIAO-ESP32C3 Version 12.1.1.6(tasmota)-2_0_5(2022.10.16 16:59:53)
00:00:00.595 WIF: verbinden mit AP1 XXXXXX Channel 10 BSSId E0:28:6D:xx:xx:xx in Modus 11n als SeeedStudioXIAOESP32C3...
00:00:02.051 WIF: verbunden
00:00:02.307 HTP: Web-Server aktiv bei SeeedStudioXIAOESP32C3 mit IP-Adresse 192.168.xxx.xxx

**Reply by barbudor** [reply]
Great to know it's working fine

May be a little typo here ?
![image](https://user-images.githubusercontent.com/2996042/196044989-8b6688f3-4bee-4658-8d76-a0082b3317a2.png)

Your logs are obviously in English

**Reply by uwe5** [reply]
Upps!
Sorry for the translation by google translation. Logs should not be translated.

**Comment by rossi42** [comment]
I have problems getting a i2c sensor connected to a esp32-c3 mini. In the console i2cscan simply replies "error". Frimware is tasmota32c3factory.bin Info shows this on console:
"Info1":{"Module":"ESP32C3","Version":"13.4.0(tasmota32)"
Any idea?

**Reply by sfromis** [reply]
If you mean that the response is `{"Command":"Error"}`, you probably did not configure the I2C bus gpio pins.

In case you tried to do that in "Configure Template", you need to have the template activated too.

When the pinout on https://wiki.seeedstudio.com/XIAO_ESP32C3_Getting_Started/ mentions SDA/SCL pins, this is merely suggestion without any hardware dependency, as ESP32 does not have specific pins for specific functions. But using gpio6/7 is fine, if you prefer, and configure Tasmota to match. Using those pins can also be a good plan if you want to use it with the Grove Base.

**Reply by rossi42** [reply]
Oh my what a horribly stupid mistake. Thank you so much. I indeed forgot to configure the SDA/SCL pins...
Now I get this:
00:00:00.029 I2C: Bus1 using GPIO09(SCL) and GPIO08(SDA)
[..]
00:00:00.112 I2C: BMP280 found at 0x76
